### PR TITLE
enable copy and paste from pre-code without + and -, which makes code to be ready to reuse

### PR DESCRIPTION
**Pull request by mhadaily**
This PR will enhance copy and paste code from documentation.

### CODE REVIEW

**Comment by mhadaily** [comment]
I am not sure why unit_tests_windows failed! tests were passed before I sent my PR.

**Comment by DSchau** [comment]
@mhadaily we're working on the windows PR failure stuff. Don't worry about it!

**Comment by stefanprobst** [comment]
Wouldn't we actually *want* some indication which lines were added/removed when copy-pasting?

**Comment by mhadaily** [comment]
@stefanjudis well I think, most of the time, you have old lines in your file or code and you want to replace it with new ones from the documentation, where it is indicated as added/removed. 

It's very likely you want to just copy the new lines, in this case, you have copied all + at the beginning of each line and need to be removed. 

This is similar to what you see in Github diff and you can easily now copy and paste without having +/-.

**Comment by fk** [comment]
(Just thinking out loud early CET morning; while I _think_ this contains all of my thoughts regarding this PR, please don't take this as a proper review/feedback. 🤗 💜)

```diff
+ I like how GitHub handles diffs — not only copying and pasting from 'em,
+ but also that + and - get their own column, and that added and removed
+ lines sport a corresponding background color.

+ I can follow both @mhadaily's as well as
- @stefanprobst's arguments,
+ probably leaning towards @mhadaily's side a little,
- but then again, looking at how what I'm currently writing here
- ends up on the clipboard … （◞‸◟）

It's complicated ¯\_(ツ)_/¯

(Btw., is it me or did GitHub change this only quite recently?
I remember being annoyed having to delete +/-
when c&p'ing from GitHub diffs not too long ago?!)

- I think the right place to implement something like this
- would be gatsby-remark-prismjs. IMO only doing so in
- template-docs-markdown does not feel right for a couple of reasons:
- Consistent UX—what if e.g. a package.json of a plugin contains diff code block?
- Sure this is an easy fix, but…
- No need to process all Markdown HTML.
- Even if it's rather unlikely, what about <a href="…">-</a> / [-](…) in Markdown?
- Just processing diff code blocks should make handling those
- edge cases a little easier.
```

**Comment by stefanprobst** [comment]
Github seems to be handling this differently depending on context:
* in a PR review pane: no +/- in selection
* in a `diff` codeblock in a comment: +/- in selection

Don't know if this is on purpose though.

**Comment by pieh** [comment]
@shannonbux did problems with copying code blocks with diff (with `-`/`+` for example like this one - https://www.gatsbyjs.org/docs/migrating-from-v1-to-v2/#5-change-query-to-use-staticquery) came up in your UX interviews? What's your take on it - should `-`/`+` be copied or not?

**Comment by shannonbux** [comment]
@pieh It seems that the `+`/`-` only exist in migration guides, because the rest of the docs and tutorials usually show additions to code examples, and rarely deletions (if ever). I just looked briefly through almost all the tutorials and there are no `+`/`-` marks. There are only highlights to emphasize new lines of code.

I haven't heard any complaints about this issue; however, it doesn't mean that it isn't a problem. I haven't done any specific studies on the migration guide.

So if this PR gets merged, it sounds like it won't affect anything except the v1 to v2 migration guide. Is that correct?

If so, I don't mind whether it gets merged or not! I'll leave it up to OSS!

File: www/src/utils/typography.js [hunk 1]
```diff
@@ -315,12 +315,21 @@ const _options = {
       ".token.atrule, .token.attr-value, .token.keyword, .token.class-name": {
         color: colors.code.keyword,
       },
+      ".token.inserted:before, .token.deleted:before": {
+        content: " ",
+      },
```
Comment (pieh) [inline www/src/utils/typography.js]: isn't this handled by 2 rules below making this no-op?

**Comment by pieh** [comment]
> Wouldn't we actually _want_ some indication which lines were added/removed when copy-pasting?

I start to lean more toward this (to keep current behaviour). If you copy diff codeblock, which has only "visual" `-`/`+` then you would need to lookup what lines are added/deleted on the page you copied it from anyway. 

I think real UX issue would be if users actually want to copy code from diff codeblock - in neither case it's truly great experience. The nice copy-worthy codeblocks should be standalone in my opinion, so maybe actual problem is that we should limit usage of diff codeblocks instead of trying to "fix" their UX?

**Comment by shannonbux** [comment]
Hm, @pieh it seems he question that would lead to a helpful answer is when
are diff code blocks useful? It sounds like they are a pain if you want to
copy and paste. They are ok if you want to just edit things yourself with
little to no copying and pasting. Is that right?

If so, yes, I'd probably recommend limiting diff code block usage to only
when it's very necessary since the majority of code blocks will have a copy
and paste button and also use highlighting to show additions. So diff code
blocks could be used to show more complicated code blocks, ones with
deletions *and* additions?

**Comment by LekoArts** [comment]
I'd also say that the `diff` block should be used in some occasions and this PR be closed to have the current behavior in these instances where you want the `- / +`

**Comment by LekoArts** [comment]
Hi @mhadaily 👋 
Sorry for the long response times. As a result of the above discussion (especially Michal's and Shannon's statements) we came to the conclusion that we should close this PR. Meaning: We want to keep the behavior that `diff` blocks let you copy the -/+ characters however limiting the usage of these blocks to specific use cases (e.g. where we want to emphasize this change and it's not necessarily a copy/paste block). Such copy/paste blocks will probably also be the main usage on the docs anyway 😊 

Thanks for opening the PR which lead to this nice discussion and a thought process about the usage of these blocks in the docs!
We look forward to seeing more PRs from you in the future! :muscle::purple_heart:
You can have a look at [our open issues][open-issues] 👍 

[open-issues]: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc

**Pull request closed without merging**